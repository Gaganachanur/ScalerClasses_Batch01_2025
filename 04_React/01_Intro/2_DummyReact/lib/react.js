function react(component){
    // Calling that function.
    console.log("..............Optimizing the changes_1");
    let dom = component();
    console.log("..............Optimizing the changes_2");

    return dom;
}