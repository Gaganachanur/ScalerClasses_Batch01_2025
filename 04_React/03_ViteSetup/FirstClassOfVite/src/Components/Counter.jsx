import React from 'react'

function COUNTER() {
    let intialState = 0;
    const [count, setCount] = React.useState(intialState);

    const increment = () => {
        setCount(count + 1);
    }

    const decrement = () => {
        setCount(count - 1);
    }

    return <div>
        <button onClick={increment}>+</button>
        <p>{count}</p>
        <button onClick={decrement}>-</button>
    </div>

}

export default COUNTER