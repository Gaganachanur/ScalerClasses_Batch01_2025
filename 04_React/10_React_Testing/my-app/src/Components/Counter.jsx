import React, { useState } from 'react'

function Counter() {
    const [count, setCount] = useState(0);

    return (
        <div className='container'>
            <button onClick={() => setCount(count + 1)}>+</button>
            <p>Count: {count}</p>
            <button onClick={() => setCount(count - 1)}>-</button>
        </div>
    )
}

export default Counter