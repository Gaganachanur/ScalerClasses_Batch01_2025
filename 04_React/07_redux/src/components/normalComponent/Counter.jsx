import React from 'react'
import { useState } from 'react'

function Counter() {
    const [count, setCounter] = useState(0);

  return (
    <div>
        <button onClick={() => setCounter(count + 1)}>+</button>
        <h3>{count}</h3>
        <button onClick={() => setCounter(count - 1)}>-</button>
    </div>
  )
}

export default Counter