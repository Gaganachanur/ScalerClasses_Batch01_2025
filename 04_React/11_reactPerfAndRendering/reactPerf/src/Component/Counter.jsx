import React from 'react'
import { useState } from 'react'

function Counter() {
    let [count, changeCount] = useState(0);

    let increment = () => {
        changeCount(count + 1);
    }

    let decrement = () => {
        changeCount(count - 1);
    }

    return (
        <>
            <button onClick={increment}>+</button>
            <p> Count: {count} </p>
            <button onClick={decrement}>-</button>

        </>
    )
}

export default Counter