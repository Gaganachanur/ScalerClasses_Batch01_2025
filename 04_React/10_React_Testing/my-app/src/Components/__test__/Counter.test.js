import { fireEvent, render, screen } from '@testing-library/react';
import Counter from '../Counter';

describe("Counter components tets cases", () => {

    test("Snapshot for counter", () => {
        const { asFragment } = render(<Counter />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("Intial State Check", () => {
        // Action.
        render(<Counter />);

        // Arrange for assertions.
        const counterText = screen.getByText("Count: 0");
        const plusText = screen.getByText("+");
        const minusText = screen.getByText("-");

        // Assertions.
        expect(counterText).toBeInTheDocument();
        expect(plusText).toBeInTheDocument();
        expect(minusText).toBeInTheDocument();
    });

    test("Increment state check", () => {
        // Action
        render(<Counter />);

        // Arrange for assertions.
        const plusText = screen.getByText("+");

        // Emulate one + click.
        fireEvent.click(plusText);

        // Assertions.
        const isOnePresent = screen.getByText("Count: 1");
        expect(isOnePresent).toBeInTheDocument();

        // Emulate one + click.
        fireEvent.click(plusText);

        // Assertions.
        const isTwoPresent = screen.getByText("Count: 2");
        expect(isTwoPresent).toBeInTheDocument();
    });

    test("Decrement state check", () => {
        // Action
        render(<Counter />);

        // Arrange for assertions.
        const minusText = screen.getByText("-");

        // Emulate one + click.
        fireEvent.click(minusText);

        // Assertions.
        const isMinusOnePresent = screen.getByText("Count: -1");
        expect(isMinusOnePresent).toBeInTheDocument();

        // Emulate one + click.
        fireEvent.click(minusText);

        // Assertions.
        const isMinusTwoPresent = screen.getByText("Count: -2");
        expect(isMinusTwoPresent).toBeInTheDocument();
    });
});