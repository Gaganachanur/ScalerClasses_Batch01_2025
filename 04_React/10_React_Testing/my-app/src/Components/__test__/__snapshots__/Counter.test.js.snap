// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter components tets cases Snapshot for counter 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button>
      +
    </button>
    <p>
      Count: 0
    </p>
    <button>
      -
    </button>
  </div>
</DocumentFragment>
`;
